Console/UI/Pages/EventsLogFilters.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Threading.Tasks;
+using Wokhan.WindowsFirewallNotifier.Console.Helpers.ViewModels;
+
+namespace Wokhan.WindowsFirewallNotifier.Console.UI.Pages
+{
+    /// <summary>
+    /// Handles filtering of security event logs
+    /// </summary>
+    internal class EventsLogFilters
+    {
+        private readonly EventsLog _eventsLog;
+
+        private static string _filterText;
+        internal string FilterText
+        {
+            get => _filterText;
+            set
+            {
+                _filterText = value;
+                ResetTextfilter();
+            }
+        }
+
+        internal EventsLogFilters(EventsLog eventsLog)
+        {
+            this._eventsLog = eventsLog;
+        }
+
+        private readonly Predicate<object> TcpFilterPredicate = (object o) =>
+        {
+            LogEntryViewModel le = (LogEntryViewModel)o;
+            bool result = le.Protocol == "TCP";
+            return result;
+        };
+
+        internal void ResetTcpFilter()
+        {
+            if (_eventsLog.dataView is null) { return; }
+
+            _eventsLog.dataView.Filter -= TcpFilterPredicate;
+            if (_eventsLog.IsTCPOnlyEnabled)
+            {
+                _eventsLog.dataView.Filter += TcpFilterPredicate;
+            }
+        }
+
+        private Predicate<object> FilterTextPredicate = (o) =>
+        {
+            LogEntryViewModel le = o as LogEntryViewModel;
+            // Note: do not use Remote Host, because this will trigger dns resolution over all entries
+            return (le.TargetIP is null ? false : le.TargetIP.StartsWith(_filterText, StringComparison.Ordinal))
```

Comment:
Shortcut : "le.TargetIP?.StartsWith(....) ?? false"

Reply:
Yes, fallback to default 😄